builtin/providers/azurerm/resource_arm_template_deployment.go

```diff
@@ -155,20 +155,44 @@ func resourceArmTemplateDeploymentRead(d *schema.ResourceData, meta interface{})
 	if resp.Properties.Outputs != nil && len(*resp.Properties.Outputs) > 0 {
 		outputs = make(map[string]string)
 		for key, output := range *resp.Properties.Outputs {
+			log.Printf("[DEBUG] Processing deployment output %s", key)
 			outputMap := output.(map[string]interface{})
 			outputValue, ok := outputMap["value"]
 			if !ok {
 				// No value
 				continue
 			}
+			outputType := outputMap["type"]
+			var outputStringValue string
+
+			switch outputType {
+			case "Bool":
+				if outputValue == false {
+					outputStringValue = "0"
+				} else if outputValue == true {
+					outputStringValue = "1"
+				} else {
+					return fmt.Errorf("Invalid value %s for boolean output %s", outputValue, key)
+				}
```

Comment:
> BTW, most of the code I see does not add [ERROR] to error messages, just to logging? What's the standard? Presumably [ERROR]'s not needed since it already shows up as an error?

Correct - apologies, I should have given a better example. It's needed on `Printf` statements, but Errors get it automatically


> Regarding Booleans, the documentation in https://www.terraform.io/docs/configuration/variables.html is a little confusing, but it looked to me like I should be explicitly using "0" and "1" instead of "false" and "true". Is that correct? If I set an output in a .tf file to be based off of a Boolean variable, I get "0" and "1" values.

Usually, we'd just rely on the `d.Set` method to handle serializing these objects into State, and not handle that directly. However given these are having to be cast to a string - I think it makes the most sense to serialize them using Go's built in `strconv.ParseBool() / strconv.FormatBool()` methods, so we can switch it out easily in the future?